src/System.Management.Automation/engine/runtime/CompiledScriptBlock.cs

```diff
@@ -1525,6 +1525,14 @@ private static bool WriteScriptBlockToLog(ScriptBlock scriptBlock, int segment,
                 // So, replace null characters with the Unicode `SYMBOL FOR NULL`
```

Comment:
@TravisEz13, your last commit had 2 failures in `PowerShell-CI-linux`
Basic SysLog tests on Linux.Verifies scriptblock logging
```powershell
Expected regular expression 'Creating Scriptblock text \(1 of 1\):#012\$pid#012& \(\[scriptblock\]::create\("Write\-Verbose 'testheader123' ;Write\-verbose 'after'"\)\)(#012)*ScriptBlock ID: [0-9a-z\-]*#012Path:.*' to match 'Creating Scriptblock text (1 of 1):#012$pid?& ([scriptblock]::create("Write-Verbose 'testheader123' ;Write-verbose 'after'"))?#012#012ScriptBlock ID: e80c2016-0469-4513-9306-fb722676c547#012Path: /tmp/138211ff-582d-4ee2-ac7b-c2f3877b7f93/test01.ps1', but it did not match.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Host/Logging.Tests.ps1: line 214
214:         $createdEvents[1].Message | Should -Match ($scriptBlockCreatedRegExTemplate -f (Get-RegEx -SimpleMatch $Script.Replace([System.Environment]::NewLine,'#012')))
```
Basic SysLog tests on Linux.Verifies scriptblock logging with null character
```powershell
Expected regular expression 'Creating Scriptblock text \(1 of 1\):#012\$pid#012& \(\[scriptblock\]::create\("Write\-Verbose 'testheader123\$\(\[char\]0x0000\)' ;Write\-verbose 'after'"\)\)(#012)*ScriptBlock ID: [0-9a-z\-]*#012Path:.*' to match 'Creating Scriptblock text (1 of 1):#012$pid?& ([scriptblock]::create("Write-Verbose 'testheader123$([char]0x0000)' ;Write-verbose 'after'"))?#012#012ScriptBlock ID: babe05f7-f77d-45a2-a574-f58c74e7bb46#012Path: /tmp/138211ff-582d-4ee2-ac7b-c2f3877b7f93/test01.ps1', but it did not match.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Host/Logging.Tests.ps1: line 243
243:         $createdEvents[1].Message | Should -Match ($scriptBlockCreatedRegExTemplate -f (Get-RegEx -SimpleMatch $Script.Replace([System.Environment]::NewLine,'#012')))
```